logger/src/main/java/com/orhanobut/logger/LoggerPrinter.java

```diff
@@ -0,0 +1,372 @@
+    private static final String TOP_BORDER = TOP_LEFT_CORNER + DOUBLE_DIVIDER + DOUBLE_DIVIDER;
+    private static final String BOTTOM_BORDER = BOTTOM_LEFT_CORNER + DOUBLE_DIVIDER + DOUBLE_DIVIDER;
+    private static final String MIDDLE_BORDER = MIDDLE_CORNER + SINGLE_DIVIDER + SINGLE_DIVIDER;
+
+    /**
+     * TAG is used for the Log, the name is a little different
+     * in order to differentiate the logs easily with the filter
+     */
+    private static String TAG = "PRETTYLOGGER";
+
+    /**
+     * Localize single tag and method count for each thread
+     */
+    private static final ThreadLocal<String> LOCAL_TAG = new ThreadLocal<>();
+    private static final ThreadLocal<Integer> LOCAL_METHOD_COUNT = new ThreadLocal<>();
+
+    /**
+     * It is used to change the tag
+     *
+     * @param tag is the given string which will be used in Logger
+     */
+    @Override
+    public Settings init(String tag) {
+        if (tag == null) {
+            throw new NullPointerException("tag may not be null");
+        }
+        if (tag.trim().length() == 0) {
+            throw new IllegalStateException("tag may not be empty");
+        }
+        LoggerPrinter.TAG = tag;
+        return settings;
+    }
+
+    @Override
+    public Printer t(String tag, int methodCount) {
+        if (tag != null) {
+            LOCAL_TAG.set(tag);
+        }
+        LOCAL_METHOD_COUNT.set(methodCount);
+        return this;
+    }
+
+    @Override
+    public void d(String message, Object... args) {
+        log(Log.DEBUG, message, args);
+    }
+
+    @Override
+    public void e(String message, Object... args) {
+        e(null, message, args);
+    }
+
+    @Override
+    public void e(Throwable throwable, String message, Object... args) {
+        if (throwable != null && message != null) {
+            message += " : " + throwable.toString();
+        }
+        if (throwable != null && message == null) {
+            message = throwable.toString();
+        }
+        if (message == null) {
+            message = "No message/exception is set";
+        }
+        log(Log.ERROR, message, args);
+    }
+
+    @Override
+    public void i(String message, Object... args) {
+        log(Log.INFO, message, args);
+    }
+
+    @Override
+    public void v(String message, Object... args) {
+        log(Log.VERBOSE, message, args);
+    }
+
+    @Override
+    public void wtf(String message, Object... args) {
+        log(Log.ASSERT, message, args);
+    }
+
+    /**
+     * Formats the json content and print it
+     *
+     * @param json the json content
+     */
+    @Override
+    public void json(String json) {
+        String tag = getTag();
+        int methodCount = getMethodCount();
+
+        if (TextUtils.isEmpty(json)) {
+            d(tag, "Empty/Null json content", methodCount);
+            return;
+        }
+        try {
+            if (json.startsWith("{")) {
```

Comment:
json string can contain empty characters and still will be valid json, so startsWith should be changed with first non empty character method